### PR TITLE
Cli 60 env config bl

### PR DESCRIPTION
**Pull request by DennyGD**
Functionality to create/update custom endpoints, collection hooks, common code, scheduled code through env config files

*Note:* IDs of modified entities aren't returned currently (except for the env id). There are only logs.

### CODE REVIEW

File: lib/EnvFileProcessor.js [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+    const envId = env.id;
+
+    async.series([
+      (next) => {
+        this.collectionsService.getAll(envId, (err, originalCollections) => {
+          if (err) {
+            return next(err);
+          }
+
+          groupedEntities = this._groupEntitiesPerOperationType(originalCollections, collections);
```
[the remark this answers is not in the document]
Comment (DennyGD) [reply lib/EnvFileProcessor.js]: Hm... how do you think they should be taken into account?

[the remark this answers is not in the document]
Comment (DennyGD) [reply lib/EnvFileProcessor.js]: When creating an env the system collections will be updated if they are defined in the config file (previously, they were just skipped).
I've removed the system collection check.
The grouping remains the same. When a decision is made for the desired behaviour, then I can make the necessary changes.

[the remark this answers is not in the document]
Comment (dokov) [reply lib/EnvFileProcessor.js]: Not good enough. If we decide to start deleting collections after 6 months, who will remember that the system collections are incorrectly marked for deletion in your object? The developer will simply uncomment the section for deletion. Same goes if someone decides to use the object for something else.

If you want to have an object like groupedEntities that holds objects for deletion, update and create, then you need to build it properly. You cannot have an entity marked for deletion and expect everyone using your object to know that he should not really delete it. This is confusing and error prone.

System collections are never deleted or created, they can only be updated, thus they may only appear in the "modified" array. You need to put this logic here. After that, everyone can use your object as is, without the need to know about system collections or anything.

[the remark this answers is not in the document]
Comment (DennyGD) [reply lib/EnvFileProcessor.js]: They are not incorrectly marked from my point of view. I was under the impression that nothing final had been decided for them. Apparently, this is not the case, so I'll make the necessary changes now.

**Pull request merged**